[MSYS] disable download timeout & overwrite

MVoz opened the pull request:
disable download timeout & overwrite & full update

TEST 
port libvpx & old version msys2

fix https://github.com/microsoft/vcpkg/issues/10937 https://github.com/microsoft/vcpkg/issues/10606

MVoz committed:
disable-download-timeout & overwrite

disable-download-timeout & overwrite & full update

TEST old version msys2 20181211

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update CONTROL

MVoz committed:
Update portfile.cmake

MVoz committed:
Update vcpkg_acquire_msys.cmake

Voskrese commented:
error: cannot remove /usr/bin/msys-2.0.dll **(Permission denied)** ?
error: cannot remove /usr/bin/pacman.exe **(Permission denied)** ?


```
2020-05-17T19:55:07.2439412Z :: Starting core system upgrade...
2020-05-17T19:55:07.2440390Z warning: resolving dependencies...
2020-05-17T19:55:07.2441418Z terminate other MSYS2 programs before proceeding
2020-05-17T19:55:07.2441978Z looking for conflicting packages...
2020-05-17T19:55:07.2442892Z 
2020-05-17T19:55:07.2444052Z Packages (8) bash-4.4.023-2  filesystem-2020.02-2  libzstd-1.4.4-2  mintty-1~3.1.4-1  msys2-runtime-3.0.7-6  pacman-5.2.1-6  pacman-mirrors-20200329-1  zstd-1.4.4-2
2020-05-17T19:55:07.2444453Z 
2020-05-17T19:55:07.2445560Z Total Download Size:   16.51 MiB
2020-05-17T19:55:07.2446373Z Total Installed Size:  65.35 MiB
2020-05-17T19:55:07.2447139Z Net Upgrade Size:      -1.76 MiB
2020-05-17T19:55:07.2447800Z 
2020-05-17T19:55:07.2448665Z :: Proceed with installation? [Y/n] 
2020-05-17T19:55:07.2449560Z :: Retrieving packages...
2020-05-17T19:55:07.2450049Z downloading msys2-runtime-3.0.7-6-x86_64.pkg.tar.xz...
2020-05-17T19:55:07.2450958Z downloading bash-4.4.023-2-x86_64.pkg.tar.xz...
2020-05-17T19:55:07.2451706Z downloading filesystem-2020.02-2-x86_64.pkg.tar.xz...
2020-05-17T19:55:07.2452641Z downloading mintty-1~3.1.4-1-x86_64.pkg.tar.xz...
2020-05-17T19:55:07.2453500Z downloading pacman-mirrors-20200329-1-any.pkg.tar.xz...
2020-05-17T19:55:07.2454351Z downloading libzstd-1.4.4-2-x86_64.pkg.tar.xz...
2020-05-17T19:55:07.2455133Z downloading zstd-1.4.4-2-x86_64.pkg.tar.xz...
2020-05-17T19:55:07.2456015Z downloading pacman-5.2.1-6-x86_64.pkg.tar.xz...
2020-05-17T19:55:07.2456743Z checking keyring...
2020-05-17T19:55:07.2457577Z checking package integrity...
2020-05-17T19:55:07.2458403Z loading package files...
2020-05-17T19:55:07.2461542Z checking for file conflicts...
2020-05-17T19:55:07.2462339Z checking available disk space...
2020-05-17T19:55:07.2462662Z :: Processing package changes...
2020-05-17T19:55:07.2462993Z upgrading msys2-runtime...
2020-05-17T19:55:07.2464181Z error: cannot remove /usr/bin/msys-2.0.dll (Permission denied)
2020-05-17T19:55:07.2464754Z warning: warning given when extracting /usr/bin/msys-2.0.dll (Could not unlink)
2020-05-17T19:55:07.2465904Z upgrading bash...
2020-05-17T19:55:07.2466911Z error: cannot remove /usr/bin/bash.exe (Permission denied)
2020-05-17T19:55:07.2467514Z warning: warning given when extracting /usr/bin/bash.exe (Could not unlink)
2020-05-17T19:55:07.2468553Z upgrading filesystem...
2020-05-17T19:55:07.2468965Z upgrading mintty...
2020-05-17T19:55:07.2469872Z upgrading pacman-mirrors...
2020-05-17T19:55:07.2471383Z installing libzstd...
2020-05-17T19:55:07.2472585Z installing zstd...
2020-05-17T19:55:07.2473419Z upgrading pacman...
2020-05-17T19:55:07.2474370Z error: cannot remove /usr/bin/pacman.exe (Permission denied)
2020-05-17T19:58:19.7138572Z warning: warning given when extracting /usr/bin/pacman.exe (Could not unlink)
2020-05-17T19:58:19.7139683Z warning: terminate MSYS2 without returning to shell and check for updates again
2020-05-17T19:58:19.7140043Z warning: for example close your terminal window instead of calling exit
```

MVoz committed:
Update portfile.cmake

MVoz committed:
Update vcpkg_acquire_msys.cmake

MVoz committed:
Update vcpkg_acquire_msys.cmake

cannot remove symbolic & junction links

Voskrese commented:
https://github.com/msys2/MSYS2-packages/issues/567

> Windows native symbolic links that are only permitted to Administrator

> fix - Win10 (developer mode enabled) ?

MVoz committed:
Enable Developer Mode

https://github.com/reupen/azure-pipelines-image-generation/blob/master/images/win/scripts/Installers/Enable-DeveloperMode.ps1

MVoz committed:
Update portfile.cmake

MVoz committed:
Update CONTROL

MVoz committed:
Update portfile.cmake

MVoz committed:
Update CONTROL

MVoz committed:
Update vcpkg_acquire_msys.cmake

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update vcpkg_acquire_msys.cmake

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update CONTROL

Voskrese commented:
example work

https://github.com/appveyor/build-images/blob/350e44f7a27e95859ac8e55bbf0e7f872153c7fa/scripts/Windows/install_msys64.ps1#L33

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update CONTROL

MVoz committed:
Update portfile.cmake

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update vcpkg_acquire_msys.cmake

MVoz committed:
Update portfile.cmake

MVoz committed:
Update CONTROL

MVoz committed:
Update portfile.cmake

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update portfile.cmake

MVoz committed:
Update CONTROL

MVoz committed:
error cannot remove (Permission denied)

error: cannot remove /usr/bin/bash.exe (Permission denied)

Voskrese commented:
> developer mode enabled - not work
> MSYS2_PATH_TYPE=inherit - not work
> MSYS=winsymlinks:nativestrict - not work

run exe Administrators ?

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update CONTROL

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update CONTROL

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update vcpkg_acquire_msys.cmake

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update CONTROL

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update vcpkg_acquire_msys.cmake

MVoz committed:
Update vcpkg_acquire_msys.cmake

MVoz committed:
Update vcpkg_acquire_msys.cmake

MVoz committed:
Update vcpkg_acquire_msys.cmake

MVoz committed:
Update vcpkg_acquire_msys.cmake

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update initialize-environment.ps1

MVoz committed:
Update vcpkg_acquire_msys.cmake

Voskrese commented:
works locally, not on the VMSS
